stream-chat-android-ui-components-sample/build.gradle

```diff
@@ -117,6 +117,14 @@ android {
     }
 }
 
+tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
+    kotlinOptions {
+        freeCompilerArgs += [
+                '-Xopt-in=io.getstream.chat.android.core.internal.InternalStreamChatApi',
```

Comment:
A few days ago I had a conversation about that with @zsmb13 and I'm not sure if we should do that in the sample app because someone could copy our `build.gradle` and they won't see any error while using our internal API. 
On the other hand, I don't have a better idea of how to solve that problem and it seems that this is still a better solution than copy-pasting or marking sample app methods with @InternalStreamChatApi annotation

Reply:
This is correct, we shouldn't need this in the sample app. The only place where using these internal APIs should come up is in the component browser, and there we should opt-in locally, for the given method or class.

Reply:
Should we perhaps split out the component browser to a new module to avoid this issue? 🤔

Reply:
Good point about copying this by customers. I think I can locally opt-in it

Reply:
If it's something that the framework calls, you can add `@InternalStreamChatApi` on that method, which technically just propagates the opt-in requirement further, but happens to work out in that case.

Otherwise you can `@OptIn(InternalStreamChatApi::class)` locally, but that needs some Gradle config too, to enable using `@OptIn` itself 😄

Reply:
I did locally `opt-in` and added a more generic compiler argument